int8 type
float32 gap_angle
